packages/run-protocol/src/vaultFactory/vaultDirector.js

```diff
@@ -74,260 +81,296 @@ const initState = (zcf, electorateParamManager, debtMint) => {
+      // totalDebt = AmountMath.add(totalDebt, toMint);
+    };
 
-      /**
-       * @param {Amount<'nat'>} toBurn
-       * @param {ZCFSeat} seat
-       */
-      const burnDebt = (toBurn, seat) => {
-        debtMint.burnLosses(harden({ RUN: toBurn }), seat);
-      };
+    /**
+     * @param {Amount<'nat'>} toBurn
+     * @param {ZCFSeat} seat
+     */
+    const burnDebt = (toBurn, seat) => {
+      debtMint.burnLosses(harden({ RUN: toBurn }), seat);
+    };
 
-      const { loanTimingParams } = zcf.getTerms();
+    const { loanTimingParams } = zcf.getTerms();
 
-      const factoryPowers = Far('vault factory powers', {
-        getGovernedParams: () => ({
-          ...vaultParamManager.readonly(),
-          getChargingPeriod: () => loanTimingParams[CHARGING_PERIOD_KEY].value,
-          getRecordingPeriod: () =>
-            loanTimingParams[RECORDING_PERIOD_KEY].value,
-        }),
-        mintAndReallocate,
-        burnDebt,
-      });
+    const factoryPowers = Far('vault factory powers', {
+      getGovernedParams: () => ({
+        ...vaultParamManager.readonly(),
+        getChargingPeriod: () => loanTimingParams[CHARGING_PERIOD_KEY].value,
+        getRecordingPeriod: () => loanTimingParams[RECORDING_PERIOD_KEY].value,
+      }),
+      mintAndReallocate,
+      burnDebt,
+    });
 
-      const vm = makeVaultManager(
-        zcf,
-        debtMint,
-        collateralBrand,
-        zcf.getTerms().priceAuthority,
-        factoryPowers,
-        timerService,
-        // @ts-expect-error
-        liquidationStrategy,
-        penaltyPoolSeat,
-        startTimeStamp,
-      );
-      collateralTypes.init(collateralBrand, vm);
-      return vm;
-    },
-    getCollaterals,
-    makeCollectFeesInvitation: makeMakeCollectFeesInvitation(
+    const vm = makeVaultManager(
+      zcf,
+      debtMint,
+      collateralBrand,
+      zcf.getTerms().priceAuthority,
+      factoryPowers,
+      timerService,
+      // @ts-expect-error
+      liquidationStrategy,
+      penaltyPoolSeat,
+      startTimeStamp,
+    );
+    collateralTypes.init(collateralBrand, vm);
+    return vm;
+  },
+  getCollaterals,
+  /** @param {MethodContext} context */
+  makeCollectFeesInvitation: ({ state }) => {
+    const { debtMint, rewardPoolSeat, zcf } = state;
+    return makeMakeCollectFeesInvitation(
       zcf,
       rewardPoolSeat,
       debtMint.getIssuerRecord().brand,
       'RUN',
-    ).makeCollectFeesInvitation,
-    getContractGovernor: () => zcf.getTerms().electionManager,
-
-    // XXX accessors for tests
-    getRewardAllocation: rewardPoolSeat.getCurrentAllocation,
-    getPenaltyAllocation: penaltyPoolSeat.getCurrentAllocation,
-  };
+    ).makeCollectFeesInvitation();
+  },
+  /** @param {MethodContext} context */
+  getContractGovernor: ({ state }) => state.zcf.getTerms().electionManager,
 
-  const machineFacet = Far('vaultFactory machine', machineBehavior);
+  // XXX accessors for tests
```

Comment:
once we have a way to instantiate a vobj from a state fixture, we don't need these methods. we can just inspect the state after the operations.